upload_vehicle_data.py

```diff
@@ -0,0 +1,327 @@
+"""This script automates the upload of rosbags from vehicle to VDI machine."""
+
+import argparse
+import logging
+import os
+import subprocess
+
+# Setup logging
+logging.basicConfig(
+    filename='upload_vehicle_data.log',
+    level=logging.INFO,
+    format='%(asctime)s %(levelname)s: %(message)s',
+)
+
+
+def run_ssh_command(remote_user, remote_ip, password, command):
+    """Run a command on the remote machine using SSH."""
+    ssh_command = (
+        f"sshpass -p {password} ssh {remote_user}@{remote_ip} '{command}'"
+    )
+    try:
+        subprocess.run(ssh_command, shell=True, check=True)
+        logging.info(f'Ran SSH command: {command}')
+    except subprocess.CalledProcessError as e:
+        logging.error(f'Failed to run SSH command: {command}: {e}')
+        raise
+
+
+def create_remote_directory(
+    remote_user, remote_ip, remote_temp_directory, password
+):
+    """Create a directory on the remote machine."""
+    command = f'mkdir -p {remote_temp_directory}'
+    run_ssh_command(remote_user, remote_ip, password, command)
+
+
+def delete_remote_directory(
+    remote_user, remote_ip, remote_temp_directory, password
+):
+    """Delete a directory on the remote machine."""
+    command = f'rm -rf {remote_temp_directory}'
+    run_ssh_command(remote_user, remote_ip, password, command)
+
+
+def compress_and_transfer_rosbag(
+    remote_user,
+    remote_ip,
+    rosbag_path,
+    remote_temp_directory,
+    vdi_upload_directory,
+    password,
+):
+    """Compress rosbag on remote machine and transfer it to VDI machine."""
+    # Compress the rosbag on the remote machine
+    remote_compressed_path = os.path.join(
+        remote_temp_directory, os.path.basename(rosbag_path)
+    )
+    mcap_path = '/home/linuxbrew/.linuxbrew/bin/mcap'
+    compress_cmd = (
+        f'{mcap_path} compress {rosbag_path} -o {remote_compressed_path}'
+    )
+    try:
+        run_ssh_command(remote_user, remote_ip, password, compress_cmd)
+        logging.info(
+            f'Compressed rosbag {rosbag_path} to'
+            f'{remote_compressed_path} on the remote machine.'
+        )
+    except subprocess.CalledProcessError as e:
+        logging.error(
+            f'Failed to compress rosbag {rosbag_path} on remote machine: {e}'
+        )
+        return False

```

Comment:
We should be careful to not run out of SSD memory while doing this. We shoudl check space memory availability  first and then try to compress. If no space is left, we should then eliminate some of the rosbags that have been already validated and uploaded.

Reply:
Added function to check space and delete the oldest mcap file.